cvat/apps/consensus/views.py

```diff
@@ -0,0 +1,474 @@
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    PartialUpdateModelMixin,
+):
+    queryset = ConsensusSettings.objects.select_related("task", "task__organization").all()
+
+    iam_organization_field = "task__organization"
+
+    search_fields = []
+    filter_fields = ["id", "task_id"]
+    simple_filters = ["task_id"]
+    ordering_fields = ["id"]
+    ordering = "id"
+
+    serializer_class = ConsensusSettingsSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.action == "list":
+            permissions = ConsensusSettingPermission.create_scope_list(self.request)
+            queryset = permissions.filter(queryset)
+
+        return queryset
+
+
+@extend_schema(tags=["consensus"])
+@extend_schema_view(
+    retrieve=extend_schema(
+        operation_id="assignee_consensus_retrieve_report",
+        summary="Get assignee consensus report details",
+        responses={
+            "200": AssigneeConsensusReportSerializer,
+        },
+    ),
+    list=extend_schema(
+        summary="List assignee consensus reports",
+        parameters=[
+            # These filters are implemented differently from others
+            OpenApiParameter(
+                "task_id", type=OpenApiTypes.INT, description="A simple equality filter for task id"
+            ),
+            OpenApiParameter(
+                "consensus_report_id",
+                type=OpenApiTypes.INT,
+                description="A simple equality filter for target",
+            ),
+        ],
+        responses={
+            "200": AssigneeConsensusReportSerializer(many=True),
+        },
+    ),
+)
+class AssigneeConsensusReportViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+):
+    queryset = AssigneeConsensusReport.objects.prefetch_related(
+        "task",
+        "task__organization",
+    ).all()
+
+    iam_organization_field = ["task__organization"]
+
+    search_fields = []
+    filter_fields = list(search_fields) + [
+        "id",
+    ]
+    simple_filters = list(set(filter_fields) - {"id"})
+    ordering_fields = list(filter_fields)
+    ordering = "id"
+
+    def get_serializer_class(self):
+        # a separate method is required for drf-spectacular to work
+        return AssigneeConsensusReportSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.action == "list":
+            if task_id := self.request.query_params.get("task_id", None):
+                # NOTE: This filter is too complex to be implemented by other means
+                try:
+                    task = Task.objects.get(id=task_id)
+                except Task.DoesNotExist as ex:
+                    raise NotFound(f"Task {task_id} does not exist") from ex
+
+                self.check_object_permissions(self.request, task)
+
+                queryset = queryset.filter(Q(task__id=task_id)).distinct()
```

Comment:
Why a custom implementation is needed here? From what I can see, it just does what a default simple filter could do.